### PR TITLE
fix(openai): add structured output instrumentation

### PR DESCRIPTION
**Pull request by 9dogs**
- [x] I have added tests that cover my changes.
- [ ] If adding a new instrumentation or changing an existing one, I've added screenshots from some observability platform showing the change.
- [x] PR name follows conventional commits format: `feat(instrumentation): ...` or `fix(instrumentation): ...`.
- [ ] (If applicable) I have updated the documentation accordingly.

Fixes https://github.com/traceloop/openllmetry/issues/1815. I'm not sure if the [response_format](https://opentelemetry.io/docs/specs/semconv/attributes-registry/gen-ai/#openai-attributes) span attribute should be added in this PR as well.

Also, the metrics tests are a bit fragile since each executed test adds its metrics to the `reader.get_metrics_data()` result, so figuring out what was added is a bit tricky.

### CODE REVIEW

File: packages/opentelemetry-instrumentation-openai/opentelemetry/instrumentation/openai/v1/__init__.py [hunk 1]
```diff
@@ -197,6 +197,32 @@ def _instrument(self, **kwargs):
                 "Assistants.create",
                 assistants_create_wrapper(tracer),
             )
+            wrap_function_wrapper(
```
Comment (nirga) [inline packages/opentelemetry-instrumentation-openai/opentelemetry/instrumentation/openai/v1/__init__.py:200]: wondering why you added these?

**Review by nirga — COMMENTED** [review]
Thanks @9dogs! We add the response format here or in a separate PR - up to you (if you do that - we might as well add everything from the recently published semantic conventions)

**Comment by 9dogs** [comment]
> Thanks @9dogs! We add the response format here or in a separate PR - up to you (if you do that - we might as well add everything from the recently published semantic conventions)

I cannot say I fully understand what's going on with semconvs, so let's leave the scope of this PR as it is.

**Review by nirga — COMMENTED** [review]
Thanks @9dogs! Left a small comment

**Review by nirga — APPROVED** [review]
Thanks @9dogs!

**Pull request merged**